modules/core/04-channel/types/upgrade.go

```diff
@@ -77,12 +77,48 @@ func (ut Timeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
 
+// UpgradeError defines an error that occurs during an upgrade.
+type UpgradeError struct {
+	// underlyingError is the underlying error that caused the upgrade to fail.
+	// this error should not be written to state.
+	underlyingError error
+	// upgradeSequence is the sequence number of the upgrade that failed.
+	upgradeSequence uint64
+}
+
+func (u *UpgradeError) Error() string {
+	return u.underlyingError.Error()
+}
+
```

Comment:
To support usage of `errormod.IsOf` could we add the following functions?
```diff
+func (u *UpgradeError) Is(err error) bool {
+       if x, ok := u.underlyingError.(interface{ Is(error) bool }); ok && x.Is(err) {
+               return true
+       }
+
+       return false
+}
+
+// Unwrap implements the built-in errors.Unwrap
+func (u *UpgradeError) Unwrap() error {
+       return u.underlyingError
+}
+
```

feel free to modify the `Is` code, I copied from stderrors library

Reply:
I think this could be nice!

Reply:
it looks like this test passes with no code changes

```golang

func (suite *KeeperTestSuite) TestUpgradeErrorIsOf() {
	ue := channeltypes.UpgradeError{}
	suite.Require().True(errorsmod.IsOf(ue, channeltypes.UpgradeError{}))
	suite.Require().False(errorsmod.IsOf(ue, channeltypes.ErrInvalidChannel))
}
```